### PR TITLE
Input method switcher

### PR DESCRIPTION
**Pull request by ksqsf**
Example profile:
```
[Groups/0]
Name=Main
Default Layout=us
DefaultIM=pinyin

[Groups/0/Items/0]
Name=rime
Layout=us

[Groups/0/Items/1]
Name=pinyin
Layout=us

[Groups/1]
Name=Other
Default Layout=us
DefaultIM=shuangpin

[Groups/1/Items/0]
Name=shuangpin
Layout=us

[Groups/1/Items/1]
Name=rime
Layout=us

[GroupOrder]
0=Main
1=Other
```

Result:
image

image

The first section (group switcher) will be hidden if there is only one group.

### CODE REVIEW

File: src/fcitx.cpp [hunk 1]
```diff
@@ -78,6 +79,7 @@ void Fcitx::setupInstance() {
     auto &addonMgr = instance_->addonManager();
     addonMgr.registerDefaultLoader(&staticAddons);
     instance_->initialize();
+    instance_->enumerate(true);
```
Comment (eagleoflqj) [inline src/fcitx.cpp:82]: Why this is needed?

File: src/controller.swift [hunk 2]
```diff
@@ -83,10 +83,71 @@ class FcitxInputController: IMKInputController {
...
+    if let sender = sender as? NSMutableDictionary {
+      if let menuItem = sender["IMKCommandMenuItem"] as? NSMenuItem {
+        let groupName = menuItem.representedObject as! String
+        set_current_input_method_group(groupName)
+      }
+    }
+  }
+
+  @objc func switchInputMethod(sender: Any?) {
+    if let sender = sender as? NSMutableDictionary {
```
Comment (eagleoflqj) [inline src/controller.swift:144]: Please extract the logic of getting `representedObject` (not necessarily a String, so the return type could be Any) to a helper function.

**Review by eagleoflqj — CHANGES_REQUESTED** [review]
Nice work, tested single and multi group.

File: src/fcitx.cpp (hunk 1 above)
Comment (ksqsf) [reply src/fcitx.cpp]: Not needed. Removed.

**Review by eagleoflqj — APPROVED** [review]

**Pull request merged**